### PR TITLE
Accepts args for port and directory, validates.  Lists server IP when…

### PR DESCRIPTION
**Pull request by saralein**
… starting.

### CODE REVIEW

**Comment by zdavy** [comment]
One overall thing that seems weird to me is that I either have to provide all of the options or none of the options, which makes them not feel like independent options.

**Comment by saralein** [comment]
Oh, you shouldn't have to provide them all.  Are you getting that impression from the ReadMe?  If so, I should fix it.  You do have to include the full "-p 6060" vs. "-p" though, but you shouldn't need "-p 6060 -d whatever" for it to run.

Or are you saying something like "-p 6066 -d" should still run with the default directory?

File: src/main/java/com/saralein/server/Constants.java [hunk 1]
```diff
@@ -7,6 +7,12 @@
 public final class Constants {
```
Comment (damonkelley) [inline src/main/java/com/saralein/server/Constants.java]: I am skeptical of `Constants`. It feels weird to have command-line argument constants next to request parsing constants. 🤔

I am not suggesting you change it, just expressing my skepticism 😄

File: src/main/java/com/saralein/server/request/Request.java [hunk 2]
```diff
@@ -9,7 +9,7 @@ public Request(HashMap<String, String> request) {
         this.request = request;
     }
 
-    public String getMethod() {
+    String getMethod() {
```
Comment (damonkelley) [inline src/main/java/com/saralein/server/request/Request.java:12]: 👏

File: src/main/java/com/saralein/server/response/FileHelper.java [hunk 3]
```diff
@@ -4,7 +4,8 @@
 import java.util.List;
 
 public interface FileHelper {
```
Comment (damonkelley) [inline src/main/java/com/saralein/server/response/FileHelper.java:6]: Be wary of `*Helper` classes. They can end up being a dumping ground for unrelated things. Be vigilant in what you put here, and be on the lookout for a more specific name.

File: src/main/java/com/saralein/server/server/Server.java [hunk 4]
```diff
@@ -42,4 +47,13 @@ public void run() {
     public void stop() {
         listening = false;
     }
+
+    private void setServerIP() {
+        try {
+            InetAddress address = InetAddress.getLocalHost();
+            serverIP = address.getHostAddress();
```
Comment (damonkelley) [inline src/main/java/com/saralein/server/server/Server.java]: Would it be possible to set these via a constructor? Doing the initial setting of an instance variable outside of the field declaration or constructor feels like a code smell to me.

File: src/main/java/com/saralein/server/validation/ArgsValidation.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
+package com.saralein.server.validation;
+
+import static com.saralein.server.Constants.*;
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+
+public class ArgsValidation {
```
Comment (damonkelley) [inline src/main/java/com/saralein/server/validation/ArgsValidation.java]: This class seems to be doing more than just argument validation. It is also parsing....

We might not need to do it now, but I would recommend you think about how this class can evolve to be more Open/Closed.

File: src/main/java/com/saralein/server/server/Server.java (hunk 4 above)
Comment (damonkelley) [reply src/main/java/com/saralein/server/server/Server.java]: (But there are exceptions -- like with the Builder Pattern)

File: src/main/java/com/saralein/server/Main.java [hunk 6]
```diff
@@ -7,18 +7,26 @@
...
+            RequestParser requestParser = new RequestParser();
+            Server server = new SetupServer(logger, runtime, router, requestParser).setup(port);
 
-        RequestParser requestParser = new RequestParser();
-
-        Server server = new SetupServer(logger, runtime, router, requestParser).setup(args);
-        server.run();
+            server.run();
+        } else {
+            logger.log("Invalid arguments: Server cannot run with given arguments.");
```
Comment (zdavy) [inline src/main/java/com/saralein/server/Main.java]: I feel like this is a little hard to decipher for an end user. They wouldn't necessarily know what they missed. A fairly typical pattern is to return a list of errors from a validate method, maybe we can leverage that pattern to give more descript errors to the user.

**Review by damonkelley — APPROVED** [review]

**Review by zdavy — APPROVED** [review]

**Pull request merged**